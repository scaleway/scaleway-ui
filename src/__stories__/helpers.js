export const showAlert = (text = 'fired!') => {
  // eslint-disable-next-line no-alert
  alert(text)
}

export const log = (text = 'fired!') => {
  // eslint-disable-next-line no-console
  console.log(text)
}
