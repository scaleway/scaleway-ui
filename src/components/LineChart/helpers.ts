import { Theme } from '@emotion/react'
import { DatumValue, Serie } from '@nivo/line'

const parse = (data?: DatumValue | null) => {
  if (typeof data === 'number') return data
  if (typeof data === 'string') return parseFloat(data)
  if (data instanceof Date) return data.getTime()

  return 0
}

export const getMaxChartValue = (preppedData?: Serie[]): number => {
  if (!preppedData?.length) return 0

  const maximum = Math.max(
    ...preppedData.map(data =>
      data.data.reduce(
        (max, { y }) => Math.max(parse(y), max),
        Number.NEGATIVE_INFINITY,
      ),
    ),
  )

  return Math.ceil(maximum + maximum * 0.1)
}

export const getMinChartValue = (preppedData?: Serie[]): number => {
  if (!preppedData?.length) return 0

  const minimum = Math.min(
    ...preppedData.map(data =>
      data.data.reduce(
        (min, { y }) => Math.min(parse(y), min),
        Number.POSITIVE_INFINITY,
      ),
    ),
  )

  return Math.floor(minimum - minimum * 0.1)
}

export const getMin = (values: number[] = []): number =>
  values.length ? Math.min(...values) : 0

export const getMax = (values: number[] = []): number =>
  values.length ? Math.max(...values) : 0

export const getAverage = (values: number[] = []): number =>
  values.length
    ? Math.round(
        (values.reduce((average, curr) => average + curr, 0) / values.length) *
          100,
      ) / 100
    : 0

export const getCurrent = (values: number[] = []): number =>
  values.length ? values[values.length - 1] : 0

export const getSelected = (
  label: string,
  index: number,
  selected: string[] = [],
): string[] => {
  const labelIndexed = label + index.toString()
  const found = selected.indexOf(labelIndexed)

  if (found > -1) {
    selected.splice(found, 1)
  } else {
    selected.push(labelIndexed)
  }

  return selected
}

export const legendColors = (theme: Theme): string[] =>
  [
    theme.colors.chartGreen,
    theme.colors.chartPurple,
    theme.colors.gray350,
  ] as string[]

export const getLegendColor = (index: number, theme: Theme): string => {
  const colors = legendColors(theme)

  return colors[index] || colors[colors.length - 1]
}
