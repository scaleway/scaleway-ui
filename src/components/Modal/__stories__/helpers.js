export const animations = [
  'fold',
  'zoom',
  'scaleUp',
  'scaleBack',
  'sketch',
  'slideTop',
  'slideBottom',
  'slideLeft',
  'slideRight',
]

export const widths = ['xxsmall', 'xsmall', 'small', 'medium', 'large']

export const placements = [
  'center',
  'top-left',
  'top',
  'top-right',
  'right',
  'bottom-right',
  'bottom',
  'bottom-left',
  'left',
]
