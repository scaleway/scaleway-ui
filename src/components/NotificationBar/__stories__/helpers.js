export default [
  'security',
  'info',
  'success',
  'warning',
  'warning-blue',
  'warning-orange',
  'alert-orange',
  'alert-red',
]
