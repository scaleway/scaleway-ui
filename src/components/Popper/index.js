import { css } from '@emotion/core'
import { transparentize } from 'polished'
import PropTypes from 'prop-types'
import React, { memo, useEffect, useRef } from 'react'
import {
  Popover,
  PopoverArrow,
  PopoverDisclosure,
  usePopoverState,
} from 'reakit/Popover'
import { theme } from '../../theme'

const variants = {
  white: css`
    background-color: ${theme.white};
    color: ${theme.black};
    fill: ${theme.white};
    box-shadow: 0 2px 5px 5px ${transparentize(0.7, theme.shadow)};
  `,
  black: css`
    background-color: ${theme.black};
    color: ${theme.white};
    fill: ${theme.black};
    box-shadow: 0 2px 5px 5px ${transparentize(0.7, theme.shadow)};
  `,
}
const popoverStyle = css`
  border-radius: 4px;
`

const Disclosure = memo(({ disclosure, popover }) => {
  const innerRef = useRef(disclosure(popover))

  return (
    <PopoverDisclosure {...popover} ref={innerRef} {...disclosure.props}>
      {disclosureProps =>
        React.cloneElement(disclosure(popover), disclosureProps)
      }
    </PopoverDisclosure>
  )
})

const Popper = memo(
  ({
    animated,
    backgroudColor,
    baseId,
    children,
    disclosure,
    hasArrow,
    hideOnClickOutside,
    modal,
    placement,
    preventBodyScroll,
    visible,
    variant,
    ...props
  }) => {
    const popover = usePopoverState({
      animated,
      baseId,
      modal,
      placement,
      visible,
    })

    useEffect(() => popover.setVisible(visible), [visible])

    return (
      <>
        {disclosure && <Disclosure popover={popover} disclosure={disclosure} />}
        <Popover
          css={[popoverStyle, variants[variant]]}
          hideOnClickOutside={hideOnClickOutside}
          preventBodyScroll={preventBodyScroll}
          {...popover}
          {...props}
        >
          {hasArrow && (
            <PopoverArrow {...popover} style={{ fill: backgroudColor }} />
          )}
          {children({ ...popover })}
        </Popover>
      </>
    )
  },
)

Popper.propTypes = {
  animated: PropTypes.number,
  backgroudColor: PropTypes.string,
  baseId: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
  disclosure: PropTypes.func,
  hasArrow: PropTypes.bool,
  hideOnClickOutside: PropTypes.bool,
  modal: PropTypes.bool,
  placement: PropTypes.oneOf([
    'auto-start',
    'auto',
    'auto-end',
    'top-start',
    'top',
    'top-end',
    'right-start',
    'right',
    'right-end',
    'bottom-end',
    'bottom',
    'bottom-start',
    'left-end',
    'left',
    'left-start',
  ]),
  preventBodyScroll: PropTypes.bool,
  visible: PropTypes.bool,
  variant: PropTypes.oneOf(Object.keys(variants)),
}

Popper.defaultProps = {
  animated: 100,
  variant: 'white',
  baseId: '',
  hasArrow: true,
  hideOnClickOutside: true,
  modal: true,
  placement: 'auto',
  preventBodyScroll: false,
  visible: false,
}

export { Popper }
