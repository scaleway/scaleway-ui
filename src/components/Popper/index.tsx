import { Theme, css } from '@emotion/react'
import styled from '@emotion/styled'
import { transparentize } from 'polished'
import PropTypes from 'prop-types'
import React, {
  FunctionComponent,
  ReactElement,
  ReactNode,
  RefObject,
  isValidElement,
  memo,
  useEffect,
  useRef,
} from 'react'
import {
  Popover,
  PopoverArrow,
  PopoverDisclosure,
  PopoverProps,
  PopoverStateReturn,
  usePopoverState,
} from 'reakit/Popover'
import { Color } from '../../theme/colors'

const buildVariant =
  ({
    bgColor,
    color,
    fill,
    shadow,
  }: {
    bgColor: string
    color: string
    fill: string
    shadow: string
  }) =>
  ({ theme }: { theme: Theme }) =>
    css`
      background-color: ${theme.colors[bgColor as Color] ?? bgColor};
      color: ${theme.colors[color as Color] ?? color};
      fill: ${theme.colors[fill as Color] ?? fill};
      box-shadow: 0 2px 5px 5px
        ${transparentize(0.7, theme.colors[shadow as Color] ?? shadow)};
    `
const variants = {
  black: buildVariant({
    bgColor: 'black',
    color: 'white',
    fill: 'black',
    shadow: 'shadow',
  }),
  white: buildVariant({
    bgColor: 'white',
    color: 'black',
    fill: 'white',
    shadow: 'shadow',
  }),
} as const

type PopoverVariant = keyof typeof variants

export const popperVariants = Object.keys(variants) as PopoverVariant[]

const variantStyles = ({
  variant,
}: {
  theme: Theme
  variant: PopoverVariant
}) => variants[variant] ?? undefined

type DisclosureParam =
  | ((popover?: Partial<PopoverStateReturn>) => ReactElement)
  | ReactElement

type DisclosureProps = {
  disclosure: DisclosureParam
  popover: Partial<PopoverStateReturn>
}

const Disclosure = ({ disclosure, popover }: DisclosureProps): JSX.Element => {
  // if you need dialog inside your component, use function, otherwise component is fine
  const target = isValidElement(disclosure) ? disclosure : disclosure(popover)
  const innerRef = useRef(target) as unknown as RefObject<HTMLButtonElement>

  return (
    <PopoverDisclosure {...popover} ref={innerRef}>
      {disclosureProps => React.cloneElement(target, disclosureProps)}
    </PopoverDisclosure>
  )
}

Disclosure.propTypes = {
  disclosure: PropTypes.func.isRequired,
  popover: PropTypes.shape({}).isRequired,
}

const MemoDisclosure = memo(Disclosure)

const StyledPopover = styled(Popover, {
  shouldForwardProp: prop => !['variant'].includes(prop.toString()),
})<{ variant: PopoverVariant; name?: string }>`
  border-radius: 4px;
  ${variantStyles}
`

export type PopperProps = Partial<PopoverProps> &
  Partial<Pick<PopoverStateReturn, 'placement'>> & {
    backgroundColor?: string
    disclosure: DisclosureParam
    hasArrow?: boolean
    variant?: PopoverVariant
    name?: string
    children?: ((popover: PopoverStateReturn) => ReactNode) | ReactNode
  }

const Popper: FunctionComponent<PopperProps> = ({
  animated = 100,
  backgroundColor,
  baseId = '',
  children,
  disclosure,
  hasArrow = true,
  hideOnClickOutside = true,
  modal = true,
  name,
  placement = 'auto',
  preventBodyScroll = false,
  variant = 'white',
  visible = false,
  ...props
}) => {
  const popover = usePopoverState({
    animated,
    baseId,
    modal,
    placement,
    visible,
  })

  const { setVisible } = popover
  useEffect(() => setVisible(visible), [setVisible, visible])

  return (
    <>
      {disclosure && (
        <MemoDisclosure popover={popover} disclosure={disclosure} />
      )}
      <StyledPopover
        name={name}
        aria-label={name}
        variant={variant}
        hideOnClickOutside={hideOnClickOutside}
        preventBodyScroll={preventBodyScroll}
        {...popover}
        {...props}
      >
        {hasArrow && (
          <PopoverArrow {...popover} style={{ fill: backgroundColor }} />
        )}
        {typeof children === 'function' ? children({ ...popover }) : children}
      </StyledPopover>
    </>
  )
}

Popper.propTypes = {
  animated: PropTypes.number,
  backgroundColor: PropTypes.string,
  baseId: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
  disclosure: PropTypes.oneOfType([PropTypes.element, PropTypes.func])
    .isRequired,
  /**
   * Determine if an arrow is added on the direction of main content.
   */
  hasArrow: PropTypes.bool,
  hideOnClickOutside: PropTypes.bool,
  modal: PropTypes.bool,
  name: PropTypes.string,
  placement: PropTypes.oneOf([
    'auto-start',
    'auto',
    'auto-end',
    'top-start',
    'top',
    'top-end',
    'right-start',
    'right',
    'right-end',
    'bottom-end',
    'bottom',
    'bottom-start',
    'left-end',
    'left',
    'left-start',
  ]),
  /**
   * Block main content / body scroll when popper is opened.
   */
  preventBodyScroll: PropTypes.bool,
  variant: PropTypes.oneOf<PopoverVariant>(
    Object.keys(variants) as PopoverVariant[],
  ),
  visible: PropTypes.bool,
}

const MemoPopper = memo(Popper)

export default MemoPopper
