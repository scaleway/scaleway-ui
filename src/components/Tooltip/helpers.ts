import { RefObject } from 'react'

export type TooltipPlacement = 'top' | 'right' | 'bottom' | 'left' | 'auto'
export const ARROW_WIDTH = 6 // in px
export const SPACE = 6 // in px
export const TOTAL_USED_SPACE = ARROW_WIDTH + SPACE // in px
export const DEFAULT_POSITIONS = {
  arrowLeft: 50,
  arrowTop: 99,
  arrowTransform: 'translate(-50%, -50)',
  left: 0,
  rotate: 135,
  tooltipInitialPosition: 'translate3d(0, 0, 0))',
  top: 0,
}

export const TOOLTIP_INITIAL_POSITION = {
  auto: `translate3d(0, ${TOTAL_USED_SPACE}px, 0)`,
  bottom: `translate3d(0, -${TOTAL_USED_SPACE}px, 0)`,
  left: `translate3d(${TOTAL_USED_SPACE}px, 0, 0)`,
  right: `translate3d(-${TOTAL_USED_SPACE}px, 0, 0)`,
  top: `translate3d(0, ${TOTAL_USED_SPACE}px, 0)`,
}

type ComputePlacementTypes = {
  childrenStructuredRef: DOMRect
  tooltipStructuredRef: DOMRect
}

/**
 * This function will find the best placement in a window for tooltip based on children position and tooltip size
 */
export const computePlacement = ({
  childrenStructuredRef,
  tooltipStructuredRef,
}: ComputePlacementTypes) => {
  const {
    top: childrenTop,
    left: childrenLeft,
    right: childrenRight,
  } = childrenStructuredRef

  const { width: tooltipWidth, height: tooltipHeight } = tooltipStructuredRef

  if (childrenTop - tooltipHeight - TOTAL_USED_SPACE < 0) {
    return 'bottom'
  }

  if (childrenLeft - tooltipWidth - TOTAL_USED_SPACE < 0) {
    return 'right'
  }

  if (childrenRight + tooltipWidth + TOTAL_USED_SPACE > window.innerWidth) {
    return 'left'
  }

  return 'top'
}

type ComputePositionsTypes = {
  placement: TooltipPlacement
  childrenRef: RefObject<HTMLDivElement>
  tooltipRef: RefObject<HTMLDivElement>
}

/**
 * This function will compute the positions of tooltip and arrow based on children position and tooltip size
 */
export const computePositions = ({
  placement,
  childrenRef,
  tooltipRef,
}: ComputePositionsTypes) => {
  const childrenStructuredRef = (
    childrenRef.current as HTMLDivElement
  ).getBoundingClientRect()
  const tooltipStructuredRef = (
    tooltipRef.current as HTMLDivElement
  ).getBoundingClientRect()

  const placementBasedOnWindowSize =
    placement === 'auto'
      ? computePlacement({
          childrenStructuredRef,
          tooltipStructuredRef,
        })
      : placement

  const {
    top: childrenTop,
    left: childrenLeft,
    right: childrenRight,
    width: childrenWidth,
    height: childrenHeight,
  } = childrenStructuredRef

  const { width: tooltipWidth, height: tooltipHeight } = tooltipStructuredRef

  switch (placementBasedOnWindowSize) {
    case 'bottom':
      return {
        arrowLeft: tooltipWidth / 2,
        arrowTop: -ARROW_WIDTH - 5,
        arrowTransform: '',
        left: childrenLeft + childrenWidth / 2 - tooltipWidth / 2,
        rotate: 180,
        tooltipInitialPosition: TOOLTIP_INITIAL_POSITION.bottom,
        top: childrenTop + childrenHeight + ARROW_WIDTH + SPACE,
      }
    case 'left':
      return {
        arrowLeft: tooltipWidth + ARROW_WIDTH + 5,
        arrowTop: tooltipHeight / 2,
        arrowTransform: 'translate(-50%, -50%)',
        left: childrenLeft - tooltipWidth - ARROW_WIDTH - SPACE * 2,
        rotate: -90,
        tooltipInitialPosition: TOOLTIP_INITIAL_POSITION.left,
        top: childrenTop - tooltipHeight / 2 + childrenHeight / 2,
      }
    case 'right':
      return {
        arrowLeft: -ARROW_WIDTH - 5,
        arrowTop: tooltipHeight / 2,
        arrowTransform: 'translate(50%, -50%)',
        left: childrenRight + ARROW_WIDTH + SPACE * 2,
        rotate: 90,
        tooltipInitialPosition: TOOLTIP_INITIAL_POSITION.right,
        top: childrenTop - tooltipHeight / 2 + childrenHeight / 2,
      }
    default: // top placement is default value
      return {
        arrowLeft: tooltipWidth / 2,
        arrowTop: tooltipHeight - 1,
        arrowTransform: '',
        left: childrenLeft + childrenWidth / 2 - tooltipWidth / 2,
        rotate: 0,
        tooltipInitialPosition: TOOLTIP_INITIAL_POSITION.top,
        top: childrenTop - tooltipHeight - ARROW_WIDTH - SPACE,
      }
  }
}
