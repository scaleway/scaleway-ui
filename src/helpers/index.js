export * from './clipboard'

export const parseIntOr = (str, fallback) => {
  const value = Number.parseInt(str, 10)
  return !Number.isNaN(value) ? value : fallback
}

export const onKeyOnlyNumbers = e => {
  // `e.which` and `e.keyCode` are deprecated (https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent)
  const keyCode = e.key.charCodeAt(0)
  if (keyCode < 48 || keyCode > 57) {
    e.preventDefault()
  }
}
