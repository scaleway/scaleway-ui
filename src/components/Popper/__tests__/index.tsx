import Popper, { popperVariants } from '..'
import { shouldMatchEmotionSnapshotWithPortal } from '../../../helpers/jestHelpers'

describe('Popper', () => {
  test(`renders with modal=false`, () =>
    shouldMatchEmotionSnapshotWithPortal(
      <Popper
        aria-label="test"
        modal={false}
        baseId="popover-test-1"
        disclosure={() => <button type="button">modal=false</button>}
      >
        {() => <div>test</div>}
      </Popper>,
    ))

  test(`renders with disclosure`, () =>
    shouldMatchEmotionSnapshotWithPortal(
      <Popper
        aria-label="Custom popover with button"
        baseId="popover-test-2"
        disclosure={<button type="button">Test</button>}
        placement="auto"
        modal={false}
      >
        {({ placement }) => <div> {placement}</div>}
      </Popper>,
    ))

  test.each(popperVariants)('renders correctly with type="%s"', variant =>
    shouldMatchEmotionSnapshotWithPortal(
      <Popper
        aria-label={variant}
        variant={variant}
        baseId={`popover-test-${variant}`}
        disclosure={() => <button type="button">{variant}</button>}
      >
        {() => <div>{variant}</div>}
      </Popper>,
    ),
  )
})
