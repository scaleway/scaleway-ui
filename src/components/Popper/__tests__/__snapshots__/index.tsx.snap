// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popper renders correctly with type="black" 1`] = `
<DocumentFragment>
  <button
    aria-controls="popover-test-black"
    aria-expanded="false"
    aria-haspopup="dialog"
    type="button"
  >
    black
  </button>
  .cache-1uoj074 {
  border-radius: 4px;
  background-color: #000;
  color: #fff;
  fill: #000;
  box-shadow: 0px 4px 24px 6px #D4DAE766;
}

<div
    class="__reakit-portal"
  >
    <div
      aria-label="black"
      aria-modal="true"
      class="cache-1uoj074 e1u0xa8a0"
      data-dialog="true"
      hidden=""
      id="popover-test-black"
      role="dialog"
      style="display: none; position: fixed; left: 100%; top: 100%;"
      tabindex="-1"
    >
      <div
        style="font-size: 30px; width: 1em; height: 1em; pointer-events: none;"
      >
        <svg
          viewBox="0 0 30 30"
        >
          <path
            class="stroke"
            d="M23.7,27.1L17,19.9C16.5,19.3,15.8,19,15,19s-1.6,0.3-2.1,0.9l-6.6,7.2C5.3,28.1,3.4,29,2,29h26 C26.7,29,24.6,28.1,23.7,27.1z"
          />
          <path
            class="fill"
            d="M23,27.8c1.1,1.2,3.4,2.2,5,2.2h2H0h2c1.7,0,3.9-1,5-2.2l6.6-7.2c0.7-0.8,2-0.8,2.7,0L23,27.8L23,27.8z"
          />
        </svg>
      </div>
      <div>
        black
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Popper renders correctly with type="white" 1`] = `
<DocumentFragment>
  <button
    aria-controls="popover-test-white"
    aria-expanded="false"
    aria-haspopup="dialog"
    type="button"
  >
    white
  </button>
  .cache-1dlx7rm {
  border-radius: 4px;
  background-color: #fff;
  color: #000;
  fill: #fff;
  box-shadow: 0px 4px 24px 6px #D4DAE766;
}

<div
    class="__reakit-portal"
  >
    <div
      aria-label="white"
      aria-modal="true"
      class="cache-1dlx7rm e1u0xa8a0"
      data-dialog="true"
      hidden=""
      id="popover-test-white"
      role="dialog"
      style="display: none; position: fixed; left: 100%; top: 100%;"
      tabindex="-1"
    >
      <div
        style="font-size: 30px; width: 1em; height: 1em; pointer-events: none;"
      >
        <svg
          viewBox="0 0 30 30"
        >
          <path
            class="stroke"
            d="M23.7,27.1L17,19.9C16.5,19.3,15.8,19,15,19s-1.6,0.3-2.1,0.9l-6.6,7.2C5.3,28.1,3.4,29,2,29h26 C26.7,29,24.6,28.1,23.7,27.1z"
          />
          <path
            class="fill"
            d="M23,27.8c1.1,1.2,3.4,2.2,5,2.2h2H0h2c1.7,0,3.9-1,5-2.2l6.6-7.2c0.7-0.8,2-0.8,2.7,0L23,27.8L23,27.8z"
          />
        </svg>
      </div>
      <div>
        white
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Popper renders with disclosure 1`] = `
<DocumentFragment>
  <button
    aria-controls="popover-test-2"
    aria-expanded="false"
    aria-haspopup="dialog"
    type="button"
  >
    Test
  </button>
  .cache-1dlx7rm {
  border-radius: 4px;
  background-color: #fff;
  color: #000;
  fill: #fff;
  box-shadow: 0px 4px 24px 6px #D4DAE766;
}

<div
    aria-label="Custom popover with button"
    class="cache-1dlx7rm e1u0xa8a0"
    data-dialog="true"
    hidden=""
    id="popover-test-2"
    role="dialog"
    style="display: none; position: fixed; left: 100%; top: 100%;"
    tabindex="-1"
  >
    <div
      style="font-size: 30px; width: 1em; height: 1em; pointer-events: none;"
    >
      <svg
        viewBox="0 0 30 30"
      >
        <path
          class="stroke"
          d="M23.7,27.1L17,19.9C16.5,19.3,15.8,19,15,19s-1.6,0.3-2.1,0.9l-6.6,7.2C5.3,28.1,3.4,29,2,29h26 C26.7,29,24.6,28.1,23.7,27.1z"
        />
        <path
          class="fill"
          d="M23,27.8c1.1,1.2,3.4,2.2,5,2.2h2H0h2c1.7,0,3.9-1,5-2.2l6.6-7.2c0.7-0.8,2-0.8,2.7,0L23,27.8L23,27.8z"
        />
      </svg>
    </div>
    <div>
       auto
    </div>
  </div>
</DocumentFragment>
`;

exports[`Popper renders with modal=false 1`] = `
<DocumentFragment>
  <button
    aria-controls="popover-test-1"
    aria-expanded="false"
    aria-haspopup="dialog"
    type="button"
  >
    modal=false
  </button>
  .cache-1dlx7rm {
  border-radius: 4px;
  background-color: #fff;
  color: #000;
  fill: #fff;
  box-shadow: 0px 4px 24px 6px #D4DAE766;
}

<div
    aria-label="test"
    class="cache-1dlx7rm e1u0xa8a0"
    data-dialog="true"
    hidden=""
    id="popover-test-1"
    role="dialog"
    style="display: none; position: fixed; left: 100%; top: 100%;"
    tabindex="-1"
  >
    <div
      style="font-size: 30px; width: 1em; height: 1em; pointer-events: none;"
    >
      <svg
        viewBox="0 0 30 30"
      >
        <path
          class="stroke"
          d="M23.7,27.1L17,19.9C16.5,19.3,15.8,19,15,19s-1.6,0.3-2.1,0.9l-6.6,7.2C5.3,28.1,3.4,29,2,29h26 C26.7,29,24.6,28.1,23.7,27.1z"
        />
        <path
          class="fill"
          d="M23,27.8c1.1,1.2,3.4,2.2,5,2.2h2H0h2c1.7,0,3.9-1,5-2.2l6.6-7.2c0.7-0.8,2-0.8,2.7,0L23,27.8L23,27.8z"
        />
      </svg>
    </div>
    <div>
      test
    </div>
  </div>
</DocumentFragment>
`;
